### DISCUSSION TITLE
Destructive behavior of RustDesk

### DISCUSSION THREAD

**Discussion by Tamila-2017** [rustdesk/rustdesk#10176] [General]
### Bug Description

I had the official Nvidia video driver for the RTX 3090 graphics card installed on Debian 12, which worked fine.
However, after installing RustDesk, this driver it stopped working, was destroyed without warning
It was terrible! I had to remove RustDesk and install the video driver NVIDIA again.
That's why I can't use destructive RustDesk.

### How to Reproduce

Install the RustDesk

### Expected Behavior

RustDesk should not destroy the installed drivers.

### Operating system(s) on local (controlling) side and remote (controlled) side

Debian 12.6

### RustDesk Version(s) on local (controlling) side and remote (controlled) side

1.2.3

### Screenshots

![RustDesk destroyer](https://github.com/user-attachments/assets/dc7bff5d-78b0-4173-9f44-3c31a4b3204b)

**Comment by xlionjuan** [comment]
1. Is apt/dpkg removing or touch anything while you installing RustDesk? If so, list it.
2. You could try install `nvidia-open` driver direct from NVIDIA Cuda Toolkit repo, make sure your `purge` all NVIDIA related things first

Note: You don't need to install huge cuda-toolkit to install `nvidia-open`

https://developer.nvidia.com/cuda-downloads?target_os=Linux&target_arch=x86_64&Distribution=Debian&target_version=12&target_type=deb_network

```bash
wget https://developer.download.nvidia.com/compute/cuda/repos/debian12/x86_64/cuda-keyring_1.1-1_all.deb
sudo dpkg -i cuda-keyring_1.1-1_all.deb
sudo add-apt-repository contrib
sudo apt-get update
sudo apt-get -y install nvidia-open
```

**Comment by Tamila-2017** [comment]
**xlionjuan**, thank you for your reply. 
I need to tell you more about a situation that looks different.
I am installing the official NVIDIA driver from the link https://www.nvidia.com/en-us/drivers / in which me need to specify the video card  are using.
As a result, I get an installer file called '**NVIDIA-Linux-x86_64-550.127.05.run**'
I run this file, answer the questions asked, and install the NVIDIA video driver, which is working successfully.
But when I install RustDesk after that, this driver **is destroyed**.
PS. Unfortunately, the nvidia-open driver is not suitable for my tasks.

**Reply by 21pages** [reply]
Disable hardware codec option.

**Reply by Tamila-2017** [reply]
**21pages**
Sorry, but this tip is probably meant for another user? Because I don't use this option.
So I don't understand what option you are talking about.

**Reply by xlionjuan** [reply]
Generally, installing `.run` format drivers **is not suggested**, if the open one isn't suitable for you, your could try the proprietary one:

```bash
sudo apt-get install -y cuda-drivers
```

> So I don't understand what option you are talking about.

It is at RustDesk setting

**Reply by xlionjuan** [reply]
`.run` method should be the **last resort**, such as using very uncommon distros, but since you're using Debian, which is the very common distro, you should only use package manager to install NVIDIA Drivers.

When using `.run` format drivers, any breaking or dirtying your system environment **is expected**, because they're outside the control of the package manager.

**Comment by Tamila-2017** [comment]
> Generally, installing .run format drivers is not suggested, 

I'm curious, who didn't recommend it? 
Anyway, .run is recommended by the NVIDIA manufacturer.

> It is at RustDesk setting

This is useless, because RustDesk destroy  the official NVIDIA driver even during his installation, and it will be too late to use disabling the "disable hardware Codec option" 

> sudo apt-get install -y cuda-drivers

I'm not sure that this method of installing the driver will allow me too to use the video card to perform artificial intelligence tasks.

But you haven't answered the main question - why does RustDesk destroy the installed video driver? Why did you implement it so poorly?

**Reply by xlionjuan** [reply]
I'm using `nvidia-open` on multiple machines and working on AI without any problems.

If you **hate** open source driver, go ahead to install the proprietary one, if you still not willing to installing the driver with package manager, ***potential system damage is expected***, I don't want to repeat again.

https://www.reddit.com/r/linux_gaming/comments/1diijul/i_get_this_error_when_trying_to_run_an_nvidia/
https://forums.rockylinux.org/t/nvidia-drivers-on-rocky-linux/12366

**Comment by Tamila-2017** [comment]
> If you hate open source driver, go ahead to install the proprietary one

On the contrary, I like open source drivers, but I couldn't use them for AI.

Thanks for your tips, I'll give them a try

**Reply by xlionjuan** [reply]
If installing using `apt`, if something broken, or you still need proprietary driver for somehow, you could `apt purge` them very easily

 `.run` format will definitely break during dist upgrade (ex. Debian 12 to 13) because package manager do not know the existence of them.

Tip: I don't know how to purge or clean the driver installation with `.run` method, please Google to find the list to cleanup, but you could try `ncdu` to assist you, it is kinda like WizTree on Windows, syntax: `sudo ncdu /`

**Reply by Tamila-2017** [reply]
> If installing using apt, if something broken, or you still need proprietary driver for somehow, you could apt purge them very easily

Yes, I am aware of this, it is a very useful  property **apt**.

> I don't know how to purge or clean the driver installation with .run method, 

I know how to do it, and I've done it before. If you need it, I can tell you how to do it.

**Reply by xlionjuan** [reply]
Thanks but I think I don't need to know, this kind of things in the Linux world are mature nowadays.

Is your driver still breaking with package manager installation?

**Reply by Tamila-2017** [reply]
Thank you. I'm currently preparing an hardware upgrade. After completing it, I will try your recommendations and tell you about the results.

**Comment by Tamila-2017** [comment]
**xlionjuan**

I installed Debian on a new hardware.
Then, on your advice, I installed an open NVIDIA driver:
```
wget https://developer.download.nvidia.com/compute/cuda/repos/debian12/x86_64/cuda-keyring_1.1-1_all.deb
sudo dpkg -i cuda-keyring_1.1-1_all.deb
sudo add-apt-repository contrib
sudo apt-get update
sudo apt-get -y install nvidia-open
```
After that, the screen resolution decreased to 1024x768 and it is impossible to set another resolution.

But the main thing is different: when the proprietary driver '**NVIDIA-Linux-x86_64-550.127.05.run**' was installed earlier
then the speed of Ollama's work with the Lama 3.1 model was **high**: **~90** tokens/sec.

Now that the open NVIDIA driver is used, the speed of Ollama is **10 times less**: **~8** tokens/sec.

This means that the open NVIDIA driver does not work with AI, the CPU works instead.

Conclusion: **the open NVIDIA driver is unsuitable for AI**.

**Reply by xlionjuan** [reply]
Purge driver and try `cuda-drivers`

**Reply by xlionjuan** [reply]
Ollama is too basic that I can't making it not work, I'm using `ghcr.io/open-webui/open-webui:ollama` container image, which is Open WebUI + Ollama bundle, and it works great, but it may not using GPU for somehow, but it can fixed by restart the container.

**Comment by Tamila-2017** [comment]
I also forgot to say that at the end of installing the open driver, for some reason, there were many errors, but it doesn't matter anymore...

![Possible missing firmware](https://github.com/user-attachments/assets/d7d55229-267b-462b-b328-0fd38794b24f)

**Comment by Tamila-2017** [comment]
Ok, I'll install the driver using this command:

`    sudo apt-get install -y cuda-drivers`

on new pure Debian,

**Reply by xlionjuan** [reply]
There is a `apt purge` command.

**Comment by Tamila-2017** [comment]
I installed the CUDA driver using this command -

`sudo apt-get install -y cuda-drivers`

and now everything is fine - the NVIDIA driver works great and Rust Desk does not destroy it!

But the driver only works if **CSM** (Compatibility Support Module) mode is set in the BIOS

If I set the **UEFI** mode, then the driver does not work at all, and there are such errors in the log -
```
 nvidia-persistenced[609]: Failed to query NVIDIA devices. Please ensure that the NVIDIA device files (/dev/nvidia*) exist, and that user>
Dec 15 18:43:07 server systemd[1]: Failed to start nvidia-persistenced.service - NVIDIA Persistence Daemon.
Dec 15 18:43:07 server nvidia-persistenced[662]: Failed to query NVIDIA devices. Please ensure that the NVIDIA device files (/dev/nvidia*) exist, and that user>
Dec 15 18:43:07 server systemd[1]: Failed to start nvidia-persistenced.service - NVIDIA Persistence Daemon.
Dec 15 18:43:08 server nvidia-persistenced[687]: Failed to query NVIDIA devices. Please ensure that the NVIDIA device files (/dev/nvidia*) exist, and that user>
Dec 15 18:43:08 server systemd[1]: Failed to start nvidia-persistenced.service - NVIDIA Persistence Daemon.
Dec 15 18:43:09 server nvidia-persistenced[703]: Failed to query NVIDIA devices. Please ensure that the NVIDIA device files (/dev/nvidia*) exist, and that user>
Dec 15 18:43:09 server systemd[1]: Failed to start nvidia-persistenced.service - NVIDIA Persistence Daemon.
Dec 15 18:43:09 server nvidia-persistenced[722]: Failed to query NVIDIA devices. Please ensure that the NVIDIA device files (/dev/nvidia*) exist, and that user>
Dec 15 18:43:09 server systemd[1]: Failed to start nvidia-persistenced.service - NVIDIA Persistence Daemon.
Dec 15 18:43:09 server systemd[1]: Failed to start nvidia-persistenced.service - NVIDIA Persistence Daemon.
```
Unfortunately, I don't understand why this is happening :-(

**Reply by xlionjuan** [reply]
Disable Secure Boot.

**Comment by Tamila-2017** [comment]
Thank you, but unfortunately, in the BIOS of the motherboard "MSI PRO B760-PDDR4II" I did not find such an option.
I can only disable Secure Boot by switching to CSM.

**Reply by xlionjuan** [reply]
https://youtu.be/EypQEavuO_8

**Comment by Tamila-2017** [comment]
Unfortunately, I was unlucky:  this option is missing in **MSI PRO B760-PDDR4 II**  for an unknown reason...

![MSI PRO B760-PDDR4II](https://github.com/user-attachments/assets/b6358c78-d19f-49b4-b59c-6f761b7237b6)

**Reply by xlionjuan** [reply]
Main Menu's right down corner I guess

Source: https://www.mobile01.com/topicdetail.php?f=488&t=6473900 (Chinese)

**Comment by Tamila-2017** [comment]
Oh yes, yes! With your help, I have revealed this  Secure Boot!  &#128512;


**xlionjuan**,  You are very kind and attentive, thank you very much! &#127867;

PS. The problem is solved, the topic can be closed

**Reply by xlionjuan** [reply]
(Only issues have "closed" status, discussion is more like forum posts.)
